pkg/config/datafileprojectconfig/mappers/condition_trees.go

```diff
@@ -93,22 +81,64 @@ func buildConditionTree(conditions interface{}) (conditionTree *entities.TreeNod
 		}
 	}
 
-	populateConditions(value, conditionTree)
+	// Check for leaf conditions
+	if value.Kind() == reflect.Map {
+		typedV := value.Interface()
+		if v, ok := typedV.(map[string]interface{}); ok {
+			n := &entities.TreeNode{}
+			if err := createLeafCondition(v, n); err != nil {
```

Comment:
Is there a need to create the node out here and pass it in by reference? Why not just have the createLeafCondition instantiate and populate the node and return it?

Reply:
createLeafCondition is just a wrapper around existing logic. I don't have any preferences here. if you return a node, you still need to return an error to find out if you need to do ```node.Item = condition ``` assignment and normal flow.

Reply:
it is mainly because `createLeafCondition` is being used on one other place aswell where reference passing is required.

Reply:
Ah in that makes sense